pycm/pycm_curve.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+"""Curve module."""
+from __future__ import division
+from .pycm_error import pycmCurveError
+from .pycm_util import threshold_func, thresholds_calc, isfloat
+from .pycm_param import *
+from .pycm_obj import ConfusionMatrix
+import numpy
+
+
+class Curve:
+    """
+    Curve class.
+    """
+
+    def __init__(self, actual_vector, prob_vector, classes, thresholds=None, sample_weight=None):
+        """
+        Init method.
+
+        :param actual_vector: actual vector
+        :type actual_vector: python list or numpy array of any stringable objects
+        :param prob_vector: vector of probabilities
+        :type prob_vector: python list or numpy array of any stringable objects
+        :param classes: ordered labels of classes
+        :type classes: list
+        :param thresholds: thresholds list
+        :type thresholds: list
+        :param sample_weight: sample weights list
+        :type sample_weight: list
+        """
+        self.data = {}
+        self.thresholds = []
+        __curve_validation__(actual_vector, prob_vector)
+        self.actual_vector = actual_vector
+        self.prob_vector = prob_vector
+        __curve_classes_handler__(self, classes)
+        __curve_thresholds_handler__(self, thresholds)
+        for c_index , c in enumerate(self.classes):
+            data_temp = {item:[] for item in CURVE_PARAMS}
+            for t in self.thresholds:
+                lambda_fun = lambda x: threshold_func(x, c_index, classes, t)
+                cm = ConfusionMatrix(actual_vector=self.actual_vector, predict_vector=self.prob_vector, threshold=lambda_fun, sample_weight=sample_weight)
+                for item in CURVE_PARAMS:
+                    data_temp[item].append(getattr(cm, item)[c])
+            self.data[c] = data_temp
+        self.plot_x_axis = None
+        self.plot_y_axis = None
+
+    def area(self):
+        pass
+
+    def plot(self):
+        pass
+
+
+def __curve_validation__(actual_vector, prob_vector):
+    """
+    Curve input validation.
+
+    :param actual_vector: actual vector
+    :type actual_vector: python list or numpy array of any stringable objects
+    :param prob_vector: vector of probabilities
+    :type prob_vector: python list or numpy array of any stringable objects
+    :return: None
+    """
+    if not isinstance(actual_vector, (list, numpy.ndarray)) or not isinstance(prob_vector, (list, numpy.ndarray)):
+        raise pycmCurveError(VECTOR_TYPE_ERROR)
+    if len(actual_vector) != len(prob_vector):
+        raise pycmCurveError(VECTOR_SIZE_ERROR)
+    if len(set(map(len, prob_vector))) != 1:
+        raise pycmCurveError(PROBABILITY_SIZE_ERROR)
+    for item in prob_vector:
+        if not all(map(isfloat, item)):
+            raise pycmCurveError(PROBABILITY_TYPE_ERROR)
+        if sum(item) != 1:
+            raise pycmCurveError(PROBABILITY_SUM_ERROR)
+
+
+def __curve_classes_handler__(curve, classes):
```

Comment:
Why did you use double dunders?
How about `_curve_classes_handler` since it is an inner-use function?